packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -31,16 +31,15 @@ export interface ISpinButtonState {
    * as active when up/down arrow is pressed
    */
   keyboardSpinDirection: KeyboardSpinDirection;
-
-  /**
-   * The calculated precision for the value.
-   */
-  precision: number;
 }
 
+export type DefaultProps = Required<
+  Pick<ISpinButtonProps, 'step' | 'min' | 'max' | 'disabled' | 'labelPosition' | 'label' | 'incrementButtonIcon' | 'decrementButtonIcon'>
+>;
+
 @customizable('SpinButton', ['theme', 'styles'], true)
 export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState> implements ISpinButton {
```

Comment:
I'm guessing if you add `public props: ISpinButtonProps & DefaultProps` it would remove the need for the explicit conversions to that type everywhere else.

Reply:
When I tried this, it broke a bunch of things like all the examples because it makes all these props required for all the users.

Reply:
I spoke with Cliff & Katherine about this & ended up with having to just do this.props as ISpinButtonProps & DefaultProps for every instance of this.props